src/robusta/cli/self_host.py

```diff
@@ -0,0 +1,180 @@
+            SUPABASE_KEY=anon_key,
+        )
+
+
+class RobustaRelay(BaseModel):
+    domain: str = ""
+    storePassword: str = ""
+    storeUser: str = "apiuser-robustarelay@robusta.dev"
+    storeUrl: str = ""
+    storeApiKey: str = ""  # anon key
+    slackClientId: str = "your-client-id"
+    slackClientSecret: str = "your-client-secret"
+    slackSigningSecret: str = "your-signing-secret"
+    syncActionAllowedOrigins: str = ""
+    provider: str = ""
+    apiNodePort: int = 30313  # api.domain
+    wsNodePort: int = 30314  # relay.domain
+
+    def __init__(self, domain: str, anon_key: str, provider: str, storePW: str):
+        super().__init__(
+            domain=domain,
+            storeUrl=f"https://db.{domain}",
+            syncActionAllowedOrigins=f"https://platform.{domain}",
+            storeApiKey=anon_key,
+            provider=provider,
+            storePassword=storePW,
+        )
+
+
+class SelfHostValues(BaseModel):
+    STATIC_IP_NAME: str = "robusta-platform-ip"
+    RELAY_PASSWORD: str = gen_secret(12)
+    RELAY_USER: str = "apiuser-robustarelay@robusta.dev"
+    DOMAIN: str = ""
+    PROVIDER: str = ""
+    # SUPABASE
+    JWT_SECRET: str = gen_secret(32)
+    JWT_EXPIRY: int = 3600
+    ANON_KEY: str = JWT.encode(
+        {"role": "anon", "iss": "supabase", "iat": 1661029200, "exp": 1818795600},
+        JWT_SECRET,
+    )
+    SERVICE_ROLE_KEY: str = JWT.encode(
+        {
+            "role": "service_role",
+            "iss": "supabase",
+            "iat": 1661029200,
+            "exp": 1818795600,
+        },
+        JWT_SECRET,
+    )
+    SUPABASE_URL: str = "http://kong:8000"  # Internal URL
+    PUBLIC_REST_URL: str = ""  ## Studio Public REST endpoint - replace this if you intend to use Studio outside of localhost
+
+    # POSTGRES
+    POSTGRES_PORT: int = 5432
+    POSTGRES_STORAGE: str = "50Gi"
+    POSTGRES_PASSWORD: str = gen_secret(12)
+
+    SITE_URL: str = ""  # callback target should point to the dash board
+    ADDITIONAL_REDIRECT_URLS: str = ""
+
+    DISABLE_SIGNUP: bool = False
+    # Email auth
+    ENABLE_EMAIL_SIGNUP: bool = True
+    ENABLE_EMAIL_AUTOCONFIRM: bool = True
+    SMTP_ADMIN_EMAIL: str = ""
+    SMTP_HOST: str = ""
+    SMTP_PORT: int = 2500
+    SMTP_USER: str = ""
+    SMTP_PASS: str = ""
+    SMTP_SENDER_NAME: str = ""
+
+    # Phone auth
+    ENABLE_PHONE_SIGNUP: bool = False
+    ENABLE_PHONE_AUTOCONFIRM: bool = False
+
+    # KONG API endpoint ports
+    KONG_HTTP_PORT: int = 8000
+    KONG_HTTP_NODE_PORT: int = 30312  # db.domain
+    KONG_HTTPS_PORT: int = 8443
+
+    enableRelay: bool = True
+    enableRobustaUI: bool = True
+
+
+app = typer.Typer()
+
+
+@app.command()
+def gen_config(
+    provider: str = typer.Option(
+        "",
+        help="Cloud host provider.",
+    ),
+    domain: str = typer.Option("", help="domain used to route the self host services."),
+):
+    """Create self host configuration file"""
+
+    if provider not in {"", "gke"}:
```

Comment:
I would use `on-prem` instead of ""

In addition, you're allowing empty `domain` which is illegal I think.
I would make it mandatory, with no default.
So you can't run without a `domain`

Reply:
sounds good